### PR TITLE
Improve support for Windows Embedded Compact

### PR DESCRIPTION
**Pull request by hugolm84**
### This pullrequest changes
This PR improves support for building OpenCV for Windows Embedded Compact 2013 and below.
I have tried to make this PR as small and concise as possible. Changes include:

* Zlib: Do not redefine `errno` and `fdopen` for WINCE800
* CMakeLists.txt:  `CMAKE_SYSTEM_NAME` is only available after `project()` have been called
* CMakeLists.txt: Added `WINCE` flag
* Filesystem: Do not define `OPENCV_HAVE_FILESYSTEM_SUPPORT` if platform is `WINCE`
* Glob: Only use W functions for WinAPI calls if platform is `WINCE`
* System: Use `NO_GETENV` to determine if `getenv` calls is supported or not
* System: Added code to get `temp-path` on WINCE

In addition: I have added `platforms/wince`:

* arm-wince toolchain
* headless overrides for CMake
* readme

### CODE REVIEW

File: platforms/wince/arm-wince-headless-overrides.cmake [hunk 1]
```diff
@@ -0,0 +1,11 @@
+if(WINCE)
```
[the remark this answers is not in the document]
Comment (hugolm84) [reply platforms/wince/arm-wince-headless-overrides.cmake]: Yes it works. Actually the WINCE variable is a cmake variable that's set to true if the CMAKE_SYSTEM_NAME is WindowsCE. This is available when the override file is used.
However, there is no need to set(WINCE true) after/during project() as this is done internally, I'll fix that.

File: platforms/wince/readme.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+-DWITH_PROTOBUF=OFF `
+-DBUILD_SHARED_LIBS=OFF `
+-DWITH_OPENEXR=OFF `
+-DWITH_TIFF=OFF `
+```
+
+## Building
+You are required to build using Unicode:
+`cmake --build . -j xx -- /p:CharacterSet=Unicode`
+where `xx`  is number of cores
```
[the remark this answers is not in the document]
Comment (hugolm84) [reply platforms/wince/readme.md]: Ok, as OpenCv requires minimum 3.1 I will just remove that option.

File: platforms/wince/arm-wince.toolchain.cmake [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    set(CMAKE_GENERATOR_TOOLSET CE800)
+endif()
+
+# Needed to make try_compile to succeed
+if(BUILD_HEADLESS)
+  set(CMAKE_USER_MAKE_RULES_OVERRIDE ${CMAKE_CURRENT_LIST_DIR}/arm-wince-headless-overrides.cmake)
+endif()
+
+if(NOT CMAKE_FIND_ROOT_PATH_MODE_PROGRAM)
+    set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
```
[the remark this answers is not in the document]
Comment (hugolm84) [reply platforms/wince/arm-wince.toolchain.cmake]: Ok! I was actually doing format in Vs code. Don't know why editorconfig did not take. I'll fix that :)

[the remark this answers is not in the document]
Comment (alalek) [reply platforms/wince/arm-wince.toolchain.cmake]: Perhaps this VSCode extension can handle that: https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig

File: platforms/wince/arm-wince-headless-overrides.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (alalek) [reply platforms/wince/arm-wince-headless-overrides.cmake]: Yes, looks like `set()` is not needed: https://gitlab.kitware.com/cmake/cmake/blob/v3.1.0/Modules/Platform/Windows.cmake#L4

Also there is another related CMake variable: [CMAKE_VS_WINCE_VERSION](https://gitlab.kitware.com/cmake/cmake/blob/v3.1.0/Modules/CMakeDetermineSystem.cmake#L120) (but it is [not documented well](https://www.mail-archive.com/cmake-developers@cmake.org/msg10661.html)).

[the remark this answers is not in the document]
Comment (hugolm84) [reply platforms/wince/arm-wince-headless-overrides.cmake]: > Also there is another related CMake variable: [CMAKE_VS_WINCE_VERSION](https://gitlab.kitware.com/cmake/cmake/blob/v3.1.0/Modules/CMakeDetermineSystem.cmake#L120) (but it is 
[not documented well](https://www.mail-archive.com/cmake-developers@cmake.org/msg10661.html)).

I will try to verify what CMAKE_VS_WINCE_VERSION means in this instance and if it is relevant for the toolchain to set it.

[the remark this answers is not in the document]
Comment (hugolm84) [reply platforms/wince/arm-wince-headless-overrides.cmake]: Fixed

File: platforms/wince/arm-wince.toolchain.cmake (hunk 3 above)
[the remark this answers is not in the document]
Comment (hugolm84) [reply platforms/wince/arm-wince.toolchain.cmake]: Fixed

File: platforms/wince/readme.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hugolm84) [reply platforms/wince/readme.md]: @alalek All issues are now fixed. Thank you for your comments!

**Pull request merged**